cpp/src/column/column.cu

```diff
@@ -174,11 +176,86 @@ __global__ void copy_offset_bitmask(bitmask_type *__restrict__ destination,
   }
 }
 
+/**---------------------------------------------------------------------------*
+ * @brief Copies the bits starting at the specified offset from a source
+ * bitmask into the destination bitmask.
+ *
+ * Bit `i` in `destination` will be equal to bit `i + offset` from `source`.
+ *
+ * @param destination The mask to copy into
+ * @param source The mask to copy from
+ * @param bit_offset The offset into `source` from which to begin the copy
+ * @param number_of_bits The number of bits to copy
+ *---------------------------------------------------------------------------**/
+template <int BLOCK_SIZE>
+__global__ void copy_offset_nullmask(bitmask_type *__restrict__ destination,
+                                     bitmask_type const *__restrict__ source,
+                                     size_type bit_offset,
+                                     size_type number_of_mask_words) {
+  constexpr size_type warp_size{32};
+  size_type destination_word_index = threadIdx.x + blockIdx.x * blockDim.x;
+  size_type source_word_index = destination_word_index + word_index(bit_offset);
+
+  if (destination_word_index < number_of_mask_words) {
+    bitmask_type curr_word = source[source_word_index];
+    bitmask_type next_word = 0;
+    if (destination_word_index + 1 < number_of_mask_words) {
+      next_word = source[source_word_index + 1];
+    }
+    bitmask_type write_word = __funnelshift_r(curr_word, next_word, bit_offset);
+    destination[destination_word_index] = write_word;
+  }
+}
+
+rmm::device_buffer copy_bitmask(column_view view, cudaStream_t stream,
```

Comment:
@kaatish is going to add a more generic version of this as a public API in `null_mask.hpp` that you can use instead. 

It'll look like:
```
rmm::device_buffer copy_bitmask(bitmask_type * mask, size_type begin_bit, size_type end_bit)
```

Reply:
What PR? If the PR is already open, I can depend on it.

Reply:
https://github.com/rapidsai/cudf/pull/3219

Reply:
Oh the misunderstanding! This is already his code that I've already merged.